core/services/cable_guy/main.py

```diff
@@ -1,54 +1,68 @@
-#!/usr/bin/env python3
-import logging
+#! /usr/bin/env python3
+import json
 import os
 import sys
 from pathlib import Path
-from typing import Any
+from typing import Any, List
 
-import connexion
-import flask
-from connexion.resolver import RestyResolver
-from waitress import serve
+import uvicorn
+from api import manager
+from api.manager import EthernetInterface
+from fastapi import Body, FastAPI, HTTPException, status
+from fastapi.responses import HTMLResponse
+from fastapi.staticfiles import StaticFiles
+from starlette.responses import Response as StarletteResponse
+
+
+class PrettyJSONResponse(StarletteResponse):
+    media_type = "application/json"
+
+    def render(self, content: Any) -> bytes:
+        return json.dumps(
+            content,
+            ensure_ascii=False,
+            allow_nan=False,
+            indent=2,
+            separators=(", ", ": "),
+        ).encode(self.charset)
 
-logging.basicConfig(level=logging.INFO)
 
 HTML_FOLDER = Path.joinpath(Path(__file__).parent.absolute(), "html")
+STATIC_FOLDER = Path.joinpath(HTML_FOLDER, "static")
+
+app = FastAPI(default_response_class=PrettyJSONResponse)
+app.mount("/static", StaticFiles(directory=str(STATIC_FOLDER)), name="static")
 
 
-def index() -> str:
+@app.get("/", response_class=HTMLResponse, summary="Cable Guy web interface")
+def index() -> Any:
+    """REST API endpoint to retrieve web application for Cable Guy."""
     index_file_path = Path.joinpath(HTML_FOLDER, "index.html")
     return open(index_file_path, "r").read()
 
 
-def resource(path: str, filename: str) -> Any:
-    real_path = Path.joinpath(HTML_FOLDER, f"static/{path}")
-    return flask.send_from_directory(real_path, filename)
+@app.get("/ethernet", response_model=List[EthernetInterface], summary="Retrieve ethernet interfaces.")
+def retrieve_interfaces() -> Any:
+    """REST API endpoint to retrieve the configured ethernet interfaces."""
+    return manager.ethernetManager.get_interfaces()
 
 
-# Flask CORS appears to be not working anymore with connexion
-# Check: https://github.com/zalando/connexion/issues/357
-def set_cors_headers_on_response(response: Any) -> Any:
```

Comment:
How are you dealing with CORS ?

Reply:
In fact I was thinking about that and forgot to ask: why do we need CORS here?